Data/DynamicInputPack.cs

```diff
@@ -933,12 +933,130 @@ private void WriteJson(string file)
             }
         }
 
-        /// <summary>
-        /// Given a texture will return a mapping between each emulated device name and map of key name to list of regions
-        /// </summary>
-        /// <param name="texture"></param>
-        /// <returns></returns>
-        private Dictionary<string, Dictionary<string, List<RectRegion>>> CollectRegionsForDevices(DynamicInputTexture texture)
+        #region JSON Read HELPERS
+        private void read_host_controls(JsonReader reader, string paht, ObservableCollection<HostDevice> hostDevices)
+        {
+            while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+            {
+                if (reader.TokenType == JsonToken.PropertyName)
+                {
+                    HostDevice Device = new HostDevice { Name = reader.Value.ToString() };
+                    while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+                    {
+                        if (reader.TokenType == JsonToken.PropertyName)
+                        {
+                            HostKey key = new HostKey();
+                            key.Name = reader.Value.ToString();
+                            key.TexturePath = Path.Combine(paht, reader.ReadAsString());
+                            Device.HostKeys.Add(key);
+                        }
+                    }
+                    hostDevices.Add(Device);
+                }
+            }
+        }
+
+        private void read_emulated_controls(JsonReader Reader, DynamicInputTexture texture)
```

Comment:
All of our functions are camel case, let's do that for t his and `read_host_controls` as well.